models/discounted-price/src/types.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  DiscountedPrice,
+  DiscountedPriceDraft,
+} from '@commercetools/platform-sdk';
+import { TReference } from '@commercetools-test-data/commons';
+import type { TBuilder } from '@commercetools-test-data/core';
+
+export type TDiscountedPriceRest = DiscountedPrice;
+
+export type TDiscountedPriceGraphql = DiscountedPrice & {
+  __typename: 'DiscountedProductPriceValue';
+  discountRef: TReference;
+};
```

Comment:
The `discount` property has a different shape for the GraphQL representation.

```suggestion
export type TDiscountedPriceGraphql = Omit<DiscountedPrice, 'discount'> & {
  __typename: 'DiscountedProductPriceValue';
  discount: TProductDiscountGraphql;
  discountRef: TReferenceGraphql<'discount'>;
};
```